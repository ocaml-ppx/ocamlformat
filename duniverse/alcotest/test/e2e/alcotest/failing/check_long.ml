(** These tests check the wrapping behaviour of [Alcotest.check]'s emitted diff. *)

open Alcotest_engine.Private.Utils

let id_case typ typ_str v1 v2 =
  Alcotest.test_case typ_str `Quick (fun () -> Alcotest.check typ typ_str v1 v2)

let test_char_list =
  let alphabet = List.init 26 (fun i -> Char.chr (0x61 + i)) in
  id_case Alcotest.(list char) "list" (alphabet @ [ 'A' ]) (alphabet @ [ 'B' ])

let test_int64_array =
  let ns = Array.init 100 Int64.of_int in
  id_case
    Alcotest.(array int64)
    "array"
    (Array.append ns Int64.[| max_int |])
    (Array.append ns Int64.[| min_int |])

type with_testables = E : 'a Alcotest.testable * 'a * 'a -> with_testables

(* Take two values and wrap them in [n]-many [Some]s. *)
let nested_option n : with_testables -> with_testables =
 fun t ->
  let rec inner = function
    | 0 -> t
    | n ->
        let (E (typ', v1', v2')) = inner (n - 1) in
        E (Alcotest.option typ', Some v1', Some v2')
  in
  inner n

let test_nested_options =
  let (E (typ, v1, v2)) = nested_option 35 (E (Alcotest.int, 1, 2)) in
  id_case typ "nested options" v1 v2

let () =
  let open Alcotest in
  run ~verbose:true __FILE__
    [ ("wrapping", [ test_char_list; test_int64_array; test_nested_options ]) ]
