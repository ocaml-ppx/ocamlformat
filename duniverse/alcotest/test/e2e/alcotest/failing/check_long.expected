Testing `test/e2e/alcotest/failing/check_long.ml'.
This run has ID `<uuid>'.

ASSERT list
  [FAIL]        wrapping          0   list.
ASSERT array
  [FAIL]        wrapping          1   array.
ASSERT nested options
  [FAIL]        wrapping          2   nested options.

┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        wrapping          0   list.                                    │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL list

   Expected: `[a; b; c; d; e; f; g; h; i; j; k; l; m; n; o; p; q; r; 
               s; t; u; v; w; x; y; z; A]'

   Received: `[a; b; c; d; e; f; g; h; i; j; k; l; m; n; o; p; q; r; 
               s; t; u; v; w; x; y; z; B]'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        wrapping          1   array.                                   │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL array

   Expected: `[|0; 1; 2; 3; 4; 5; 6; 7; 8; 9; 10; 11; 12; 13; 14; 15; 
                16; 17; 18; 19; 20; 21; 22; 23; 24; 25; 26; 27; 28; 29; 
                30; 31; 32; 33; 34; 35; 36; 37; 38; 39; 40; 41; 42; 43; 
                44; 45; 46; 47; 48; 49; 50; 51; 52; 53; 54; 55; 56; 57; 
                58; 59; 60; 61; 62; 63; 64; 65; 66; 67; 68; 69; 70; 71; 
                72; 73; 74; 75; 76; 77; 78; 79; 80; 81; 82; 83; 84; 85; 
                86; 87; 88; 89; 90; 91; 92; 93; 94; 95; 96; 97; 98; 99;
                9223372036854775807|]'

   Received: `[|0; 1; 2; 3; 4; 5; 6; 7; 8; 9; 10; 11; 12; 13; 14; 15; 
                16; 17; 18; 19; 20; 21; 22; 23; 24; 25; 26; 27; 28; 29; 
                30; 31; 32; 33; 34; 35; 36; 37; 38; 39; 40; 41; 42; 43; 
                44; 45; 46; 47; 48; 49; 50; 51; 52; 53; 54; 55; 56; 57; 
                58; 59; 60; 61; 62; 63; 64; 65; 66; 67; 68; 69; 70; 71; 
                72; 73; 74; 75; 76; 77; 78; 79; 80; 81; 82; 83; 84; 85; 
                86; 87; 88; 89; 90; 91; 92; 93; 94; 95; 96; 97; 98; 99;
                -9223372036854775808|]'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        wrapping          2   nested options.                          │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL nested options

   Expected: `Some
                Some
                  Some
                    Some
                      Some
                        Some
                          Some
                            Some
                              Some
                                Some
                                  Some
                                    Some
                                      Some
                                        Some
                                          Some
                                            Some
                                              Some
                                                Some
                                                  Some
                                                    Some
                                                      Some
                                                        Some
                                                          Some
                                                            Some
                                                              Some
                                                                Some
                                                                  Some
                                                                    Some
                                                                    Some
                                                                    Some
                                                                    Some
                                                                    Some
                                                                    Some
                                                                    Some
                                                                    Some 1'

   Received: `Some
                Some
                  Some
                    Some
                      Some
                        Some
                          Some
                            Some
                              Some
                                Some
                                  Some
                                    Some
                                      Some
                                        Some
                                          Some
                                            Some
                                              Some
                                                Some
                                                  Some
                                                    Some
                                                      Some
                                                        Some
                                                          Some
                                                            Some
                                                              Some
                                                                Some
                                                                  Some
                                                                    Some
                                                                    Some
                                                                    Some
                                                                    Some
                                                                    Some
                                                                    Some
                                                                    Some
                                                                    Some 2'
 ──────────────────────────────────────────────────────────────────────────────

3 failures! in <test-duration>s. 3 tests run.
